Gems/Meshlets/Code/Source/Meshlets/MeshletsAssets.cpp

```diff
@@ -0,0 +1,547 @@
+                genVtx.pz = positions[posIdx+2];
+                genVtx.nx = normals[posIdx];
+                genVtx.ny = normals[posIdx + 1];
+                genVtx.nz = normals[posIdx + 2];
+                genVtx.tx = texCoords[uvIdx];
+                genVtx.ty = texCoords[uvIdx + 1];
+            }
+
+            mesh.indices.resize(idxNum);
+            if (IndexStreamFormat == RHI::Format::R16_UINT)
+            {   // 16 bits index format 0..64K vertices
+                for (uint32_t idx = 0; idx < idxNum; ++idx)
+                {
+                    mesh.indices[idx] = (unsigned int)indices[idx];
+                }
+            }
+            else
+            {   // simple memcpy since elements are of 4 bytes size
+                memcpy(mesh.indices.data(), indices, idxNum * 4);
+            }
+
+            static bool createMeshlets = true;
+            uint32_t meshletsAmount = 0;
+            if (createMeshlets)
+            {
+                meshletsAmount = CreateMeshlets(mesh);
+            }
+            else
+            {
+                debugMarkVertexUVs(mesh);
+            }
+
+            // Copy back the altered UVs for visual verification
+            for (uint32_t vtx = 0, uvIdx = 0; vtx < vtxNum; ++vtx, uvIdx += 2)
+            {
+                GeneratorVertex& genVtx = mesh.vertices[vtx];
+
+                texCoords[uvIdx] = genVtx.tx;
+                texCoords[uvIdx + 1] = genVtx.ty;
+            }
+
+            return meshletsAmount;
+        }
+
+        uint32_t MeshletsModel::CreateMeshletsFromModelAsset(Data::Asset<RPI::ModelAsset> sourceModelAsset)
+        {
+            uint32_t meshletsAmount = 0;
+            m_sourceModelAsset = sourceModelAsset;
+
+            for (const Data::Asset<RPI::ModelLodAsset>& lodAsset : sourceModelAsset->GetLodAssets())
+            {
+                for (const RPI::ModelLodAsset::Mesh& meshAsset : lodAsset->GetMeshes())
+                {
+                    meshletsAmount += CreateMeshletsModel(meshAsset);
+                }
+            }
+
+            AZ_Warning("Meshlets", false, "Meshlet model [%s] was created", m_name.c_str());
+
+            return meshletsAmount;
+        }
+
+        uint32_t MeshletsModel::CreateMeshletsFromModel(Data::Instance<RPI::Model> sourceModel)
+        {
+            m_sourceModelAsset = sourceModel->GetModelAsset();
+            for (uint32_t lodIdx = 0; lodIdx < sourceModel->GetLodCount(); ++lodIdx)
+            {
+                const Data::Instance<RPI::ModelLod>& currentLod = sourceModel->GetLods()[lodIdx];
+
+                for (uint32_t meshIdx = 0; meshIdx < currentLod->GetMeshes().size(); ++meshIdx)
+                {
+                    [[maybe_unused]]const RPI::ModelLod::Mesh& mesh = currentLod->GetMeshes()[meshIdx];
+                    // The next is TBD - harder to get than from the reflected Asset part
+
+
+                }
+            }
+            return 0;
```

Comment:
Could this use `CreateMeshletsFromModelAsset`? 
```suggestion
            return CreateMeshletsFromModelAsset(sourceModel->GetModelAsset);
```

Reply:
Will remove this method. Wanted to keep it as reference to developers how two paths can be chosen when reconstructing, but it'll draw too many remarks.;